src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts

```diff
@@ -475,26 +477,49 @@ export class SQLFuture extends Disposable implements FutureInternal {
 
 	private async processResultSets(batch: BatchSummary): Promise<void> {
 		try {
+			let queryRowsPromises: Promise<void>[] = [];
 			for (let resultSet of batch.resultSetSummaries) {
 				let rowCount = resultSet.rowCount > this.configuredMaxRows ? this.configuredMaxRows : resultSet.rowCount;
 				if (rowCount === this.configuredMaxRows) {
 					this.handleMessage(localize('sqlMaxRowsDisplayed', "Displaying Top {0} rows.", rowCount));
 				}
-				await this.sendResultSetAsIOPub(rowCount, resultSet);
+				queryRowsPromises.push(this.getAllQueryRows(rowCount, resultSet));
+			}
```

Comment:
So one thing to note is that this is changing behavior - before if the first batch encountered an error none of the others would execute. Now we have the potential for the later ones to still execute and send their results before an error occurs. Have you tested this scenario to see if it breaks anything? I don't inherently see an issue with it as long as the caller expects that to possibly happen but let's make sure that's the case.